### PR TITLE
Set virtual project directories for `:bom`, `:version-catalog`

### PR DESCRIPTION
**Pull request by sh-cho**
Motivation:

Starting with Gradle 9.0, Gradle will not run builds if a project directory is missing or read-only.
(ref:
https://docs.gradle.org/8.7/userguide/upgrading_version_8.html#deprecated_missing_project_directory)

Modification:

- Create project directories under `build/virtual-projects`
  - `:bom`, `:version-catalog`
  - `:dependencyManagement` will be handled via https://github.com/line/gradle-scripts/pull/177

Result:

- Deprecation for Gradle 9.0 is prepared for this behavior

### CODE REVIEW

**Review by jrhee17 — APPROVED** [review]
Verified with the warning message with `./gradlew publishToMavenLocal --parallel --warning-mode all`.

A warning message for `dependencyManagement` is also printed, but I prefer that it be handled when we actually upgrade the gradle version. Thanks @sh-cho 👍 👍 👍

File: settings.gradle [hunk 1]
```diff
@@ -78,8 +78,10 @@ apply from: "${rootDir}/gradle/scripts/settings-flags.gradle"
 
 // Published BOM projects
 includeWithFlags ':bom',                                 'bom'
+project(':bom').projectDir.mkdirs()
 // Published version catalog projects
 includeWithFlags ':version-catalog',                     'version-catalog'
+project(':version-catalog').projectDir.mkdirs()
```
Comment (ikhoon) [inline settings.gradle:84]: - Should we create these projects under `gradle/virtual-projects`?
- Add `gradle/virtual-projects/` to `.gitignore`?

Comment (trustin) [reply settings.gradle:84]: Maybe `gradle/scripts/.virtual-projects/`? I wonder if it makes sense to use `build/virtual-projects` as well.. :thinking:

Comment (ikhoon) [reply settings.gradle:84]: `build/virtual-projects` sounds good because `build/` is already excluded from git and is automatically cleaned.

Comment (sh-cho) [reply settings.gradle:84]: Sounds good.  I'll try to move generated projects under `build/virtual-projects`.

Comment (sh-cho) [reply settings.gradle]: Done :)
Please take a look

**Review by minwoox — APPROVED** [review]
Thanks a lot! 👍 👍 👍 
Let me merge this after build with Gradle 9.0, which isn't release yet, to make sure if we don't miss anything.

**Review by trustin — APPROVED** [review]
Thanks, @sh-cho! :bow:

**Review by ikhoon — APPROVED** [review]
Thanks, @sh-cho!

**Comment by minwoox** [comment]
Thanks a lot, @sh-cho! 👍 👍 👍

**Pull request merged**